purchase_sale_inter_company/views/purchase_view.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field name="intercompany_sale_id" invisible="True" />
+            </field>
+            <field name="order_line" position="attributes">
+                <attribute name="attrs" operation="python_dict" key="readonly">
+                    ['|', ('state', 'in', ('done', 'cancel')), ('intercompany_sale_id', '!=', False), ('state', '=', 'purchase')]
```

Comment:
**issue**: For both logical correctness and readability, I'd change this to

```suggestion
                    ['|', 
                            ('state', 'in', ('done', 'cancel')),
                             '&nbsp;' 
                            ('intercompany_sale_id', '!=', False), 
                            ('state', '=', 'purchase')]
```

Translated: Always make it readonly during states `done` and `cancel`. If there is an `intercompany_sale_id`, make it readonly during `purchase` as well.